cheroot/test/test_conn.py

```diff
@@ -121,7 +125,39 @@ def _timeout(req, resp):
     wsgi_server.timeout = timeout
     wsgi_server.server_client = wsgi_server_client
     wsgi_server.keep_alive_conn_limit = 2
-    return wsgi_server
+
+    # patch the error_log calls of the server instance
+    class ErrorLog:  # noqa: D200
+        """
+        Mock class to access the server error_log calls made by the server.
+        """
+
+        FuncCall = namedtuple('ErrorLogCall', ['msg', 'level', 'traceback'])
+
+        def __init__(self):
+            self.calls = []
+            # to be used the the teardown validation
+            self.ignored_msgs = []
+
+        def __call__(self, msg='', level=logging.INFO, traceback=False):
+            if traceback:
+                tblines = traceback_.format_exc()
+            else:
+                tblines = ''
+            self.calls.append(ErrorLog.FuncCall(msg, level, tblines))
+
+    monkeypatch.setattr(wsgi_server, 'error_log', ErrorLog())
+    yield wsgi_server
+    # Teardown verification, in case that the server logged an
+    # error that wasn't notified to the client or we just made a mistake.
+    for c in wsgi_server.error_log.calls:
+        if c.level > logging.WARNING:
+            if c.msg not in wsgi_server.error_log.ignored_msgs:
+                raise AssertionError(
+                    'Found error in the error log: '
+                    "message = '{}', level = '{}'\n"
+                    '{}'.format(c.msg, c.level, c.traceback),
+                )
```

Comment:
1) this level of nesting is too deep & unnecessary
2) it's better to use actual `assert` statement instead of raising the error manually:
```suggestion
        if c.level <= logging.WARNING:
            continue

        assert c.msg in wsgi_server.error_log.ignored_msgs, (
            'Found error in the error log: '
            "message = '{c.msg!s}', level = '{c.level!s}'\n"
            '{c.traceback!s}'.format(**locals()),
        )
```